kernel/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+#![no_std]
+
+use core::panic::PanicInfo;
+
+#[panic_handler]
+#[no_mangle]
+pub fn panic(_info: &PanicInfo) -> ! {
+    loop {}
+}
+
+#[no_mangle]
+pub extern "C" fn rust_main() -> ! {
```

Comment:
Nitpicking, but `rust_main` doesn't seem special to anyone but us. Probably something like `kmain` would be better.